### ISSUE TITLE
ENH: scipy.integrate.quad, adding a case that if upper and lower integral bounds are the same then return zero

### ISSUE THREAD

**Issue by Shengyis** [scipy/scipy#18142]
### Is your feature request related to a problem? Please describe.

I know for normal integration with the same upper and lower bounds, quad will return zero without any problem. However, if it is an improper integral, sometimes it will give ZeroDivisionError. For example, I want to calculate \int_0^0 1/x^0.5 dx, the python code is 

def f(x):
    return x**(-0.5)
print(quad(f, 0, 0))

this will give ZeroDivisionError. However, in the view of math, since it is integrable neasr x = 0, this should return 0. 

### Describe the solution you'd like.

maybe just a few lines in quad function to judge whether two bounds are the same. If they are the same directly return 0 with err = 0, otherwise following the original quad.

**Comment by Shengyis** [comment]
![Screenshot 2023-03-13 120855](https://user-images.githubusercontent.com/72622897/224642729-ab9c2aac-fabd-427c-94de-a8a46efba151.png)

**Comment by Kai-Striega** [comment]
I think this is a good idea, but don't have the time to work on it right now. 

@Shengyis  are you thinking of working on this? I think the correct place to add this would be here:

https://github.com/scipy/scipy/blob/0a017f2f79cfd27f7d7d59a59ed759b00174f737/scipy/integrate/_quadpack_py.py#L438-L439

A test for this would need to be added.

**Comment by Shengyis** [comment]
thanks! I can try to do it. It is my first time donig this, should I request a new pull?

**Comment by Kai-Striega** [comment]
Yes, that would be great. There's more information for contributors in the [SciPy contributor guide](https://docs.scipy.org/doc/scipy/dev/contributor/contributor_toc.html) which should be helpful to get you started. If you have any more questions, feel free to reach out and I'll be happy to help!

**Comment by Shengyis** [comment]
Many thanks!

**Comment by Kai-Striega** [comment]
@Shengyis please open a PR in *addition* to the issue. The issue tells us what's wrong, the pull request is there to fix it

**Comment by Shengyis** [comment]
#18147 
This is the PR

**Comment by JostMigenda** [comment]
I discovered this issue during the EuroSciPy hack day. This still looks useful, so I’ll resurrect the previous PR.
